modules/infinite-scroll/infinity.js

```diff
@@ -247,166 +365,177 @@
 		}
 
 		// Fire the ajax request.
-		jqxhr = $.post( infiniteScroll.settings.ajaxurl, query );
+		xhr = new XMLHttpRequest();
+		xhr.open( 'POST', infiniteScroll.settings.ajaxurl, true );
+		xhr.setRequestHeader( 'Content-Type', 'application/x-www-form-urlencoded; charset=UTF-8' );
+		xhr.send( self.urlEncodeJSON( query ) );
 
 		// Allow refreshes to occur again if an error is triggered.
-		jqxhr.fail( function() {
+		xhr.onerror = function() {
 			if ( self.click_handle ) {
-				loader.hide();
+				loader.parentNode.removeChild( loader );
 			}
 
 			self.ready = true;
-		} );
+		};
 
 		// Success handler
-		jqxhr.done( function( response ) {
+		xhr.onload = function() {
+			var response = JSON.parse( xhr.responseText ),
+				httpCheck = xhr.status >= 200 && xhr.status < 300,
```

Comment:
**Note:** It's fine to check for `2xx` response codes only. All redirects are followed by `xhr` and should never be exposed to us.